akka-typed/src/main/scala/akka/typed/Behavior.scala

```diff
@@ -275,6 +277,7 @@ object Behavior {
 
   private def interpret[T](behavior: Behavior[T], ctx: ActorContext[T], msg: Any): Behavior[T] =
     behavior match {
+      case null ⇒ throw new InvalidMessageException("[null] is not an allowed message")
```

Comment:
this is matching on the `behavior` not the message

Reply:
D'oh!